packages/frontend/src/hooks/useCustomVisualizationConfig.ts

```diff
@@ -0,0 +1,51 @@
+import { ApiQueryResults, CustomVis, ResultRow } from '@lightdash/common';
+import { useMemo, useState } from 'react';
+
+const convertRowsToSeries = (rows: ResultRow[]) => {
+    return rows.map((row) => {
+        return Object.fromEntries(
+            Object.entries(row).map(([key, rowValue]) => [
+                key,
+                rowValue.value.raw,
+            ]),
+        );
+    });
+};
+
+export interface CustomVisualizationProps {
+    validConfig: CustomVis;
+    setVisSpec: (spec: string) => void;
```

Comment:
if this is a vega spec we can use type from vega? I think?

Reply:
It's not exactly. It's the vega spec but it's missing a couple things. Most notably 'data', which is required in vega. We inject that prop before passing this to Vega, but it makes sense not to store that prop at least. 

Good point about types though. Let me see if it's reasonably straight forward to use the vega type with a couple Omits or something.